test/unit/domain_constraints_test.rb

```diff
@@ -83,6 +83,13 @@ class ProviderDomainConstraintTest < DomainConstraintsTest
       assert Account.exists?(self_domain: self_domain)
       assert ProviderDomainConstraint.matches?(request)
     end
+
+    test 'master domain' do
+      master = master_account
+      request = mock
+      request.expects(:host).returns(master.domain)
+      refute ProviderDomainConstraint.matches?(request)
+    end
```

Comment:
Are you sure this goes in this PR?

Reply:
it can be in a different PR.. will split it